server/src/main/java/org/opensearch/action/search/SearchPhase.java

```diff
@@ -69,11 +70,15 @@ public String getName() {
     }
 
     /**
-     * Returns the SearchPhase name as {@link SearchPhaseName}. Exception will come if SearchPhase name is not defined
-     * in {@link SearchPhaseName}
-     * @return {@link SearchPhaseName}
+     * Returns an Optional of the SearchPhase name as {@link SearchPhaseName}. If there's not a matching SearchPhaseName,
+     * returns an empty Optional.
+     * @return {@link Optional<SearchPhaseName>}
      */
-    public SearchPhaseName getSearchPhaseName() {
-        return SearchPhaseName.valueOf(name.toUpperCase(Locale.ROOT));
+    public Optional<SearchPhaseName> getSearchPhaseName() {
```

Comment:
This is a breaking change to a class annotated `PublicApi` so I think you'll want to make this change in way that keeps the existing signature.